v2/header.go

```diff
@@ -51,3 +56,52 @@ func XGoogHeader(keyval ...string) string {
 	}
 	return buf.String()[1:]
 }
+
+// version is a package internal global variable for testing purposes.
+var version = runtime.Version
+
+// VersionUnknown is only used when the runtime version cannot be determined.
+const VersionUnknown = "UNKNOWN"
+
+// GoVersion returns a Go runtime version derived from the runtime environment
+// that is modified to be suitable for reporting in a header, meaning it has no
+// whitespace. If it is unable to determine the Go runtime version, it returns
+// VersionUnknown.
+func GoVersion() string {
```

Comment:
I am not sure if any of this is all that expensive, but should we just cache this to be safe? It could even be a var instead of a function in that case.

Reply:
Like so:
```go
var GoVersion string

func init() {
  GoVersion = goVersion()
}
```
Is that what you mean?

I'm down with that

Reply:
yup!

Reply:
Done